src/Configuration/Dotenv/Scheme.hs

```diff
@@ -10,60 +10,69 @@
 -- Helpers for loadSafeFile
 
 module Configuration.Dotenv.Scheme
-  ( checkConfig
+  ( SchemaErrors(..)
+  , checkConfig
   , checkScheme
   , readScheme
+  , defaultValidatorMap
   )
   where
 
 import Control.Monad
+import Control.Exception
 
+import Data.Typeable
 import Data.List
 import Data.Yaml (decodeFileEither, prettyPrintParseException)
 
 import Configuration.Dotenv.Scheme.Helpers
 import Configuration.Dotenv.Scheme.Parser
 import Configuration.Dotenv.Scheme.Types
 
+
+data SchemaErrors =
```

Comment:
@CristhianMotoche just `SchemaError` perhaps, since it is a sum type, therefore you are only able to use one constructor at the time. What do you think?